WordPress/gutenberg
# Extract utilities to `@wordpress/editor` and `@wordpress/router`

## What?

This PR does two things:

- Updates the duplicate action to take the the post fields from a shared place instead of duplicating it.
- Updates the view action to respect the `wp_theme_preview` parameter.

## How?

- Moves the fields declaration to the `@wordpress/editor` package, so both edit-post (actions) and edit-site (quick edit) packages have access to it.
- Moves some routing utilities to the `@wordpress/router` package, so both edit-post (actions) and edit-site (quick edit) packages have access to it.

## Testing Instructions

Test the view action:

- Visit the Site Editor's Pages page.
- Add the `wp_theme_theme=twentytwentytwo` URL parameter and reload. This should load TT2 in the preview.
- Click on the `View` action of any page, to load the front-end.
- The expected result is that the front-end loads the TT2 theme (in `trunk` the `wp_theme_preview` parameter is not used by the view action).

Test the duplicate action:

- It should still work as before, nothing changed.
